packages/manager/modules/vrack/src/dashboard/vrack.controller.js

```diff
@@ -264,82 +267,417 @@ export default class VrackMoveDialogCtrl {
+      this.updateDedicatedCloudServiceInfo();
+    }
+  }
+
+  updateDedicatedServer(dedicatedServer) {
+    if (dedicatedServer) {
+      // Update dedicated server
+      if (
+        !this.data.eligibleServices?.dedicatedServer?.length !==
+        dedicatedServer.length
+      ) {
+        this.data.eligibleServices.dedicatedServer = dedicatedServer;
+
+        // Update dedicated server data
+        this.updateDedicatedServerServiceInfo();
+      }
+    } else if (this.data.eligibleServices?.dedicatedServer?.length > 0) {
+      this.updateDedicatedServerServiceInfo();
+    }
+  }
+
+  updateDedicatedServerInterface(dedicatedServerInterface) {
+    if (dedicatedServerInterface) {
+      // Update dedicated server interface
+      if (
+        !this.data.eligibleServices?.dedicatedServerInterface?.length !==
+        dedicatedServerInterface.length
+      ) {
+        this.data.eligibleServices.dedicatedServerInterface = dedicatedServerInterface;
+
+        // Update dedicated server interface
+        this.updateDedicatedServerInterfaceServiceInfo();
+      }
+    } else if (
+      this.data.eligibleServices?.dedicatedServerInterface?.length > 0
+    ) {
+      this.updateDedicatedServerInterfaceServiceInfo();
+    }
+  }
+
+  updateIp(ip) {
+    if (ip) {
+      // Update ip
+      if (!this.data.eligibleServices?.ip?.length !== ip.length) {
+        this.data.eligibleServices.ip = ip;
+
+        // Update IP service
+        this.updateIpServiceInfo();
+      }
+    } else if (this.data.eligibleServices?.ip?.length > 0) {
+      this.updateIpServiceInfo();
+    }
+  }
+
+  updateIpv6(ipv6) {
+    if (ipv6) {
+      // Update ipv6
+      if (!this.data.eligibleServices?.ipv6?.length !== ipv6.length) {
+        this.data.eligibleServices.ipv6 = ipv6;
+
+        // Update IPv6
+        this.updateIpv6ServiceInfo();
+      }
+    } else if (this.data.eligibleServices?.ipv6?.length > 0) {
+      this.updateIpv6ServiceInfo();
+    }
+  }
+
+  updateLegacyVrack(legacyVrack) {
+    if (legacyVrack) {
+      // Update legacy vRack
+      if (
+        !this.data.eligibleServices?.legacyVrack?.length !== legacyVrack.length
+      ) {
+        this.data.eligibleServices.legacyVrack = legacyVrack;
+
+        // Update legacy vRack
+        this.updateLegacyVrackServiceInfo();
+      }
+    } else if (this.data.eligibleServices?.legacyVrack?.length > 0) {
+      this.updateLegacyVrackServiceInfo();
+    }
+  }
+
+  updateIpLoadbalancing(ipLoadbalancing) {
+    if (ipLoadbalancing) {
+      // Update load balancing
+      if (
+        !this.data.eligibleServices?.ipLoadbalancing?.length !==
+        ipLoadbalancing.length
+      ) {
+        this.data.eligibleServices.ipLoadbalancing = ipLoadbalancing;
+
+        // Update IP Loadbalancing data
+        this.updateIpLoadbalancingServiceInfo();
+      }
+    } else if (this.data.eligibleServices?.ipLoadbalancing?.length > 0) {
+      this.updateIpLoadbalancingServiceInfo();
+    }
+  }
```

Comment:
**suggestion:** The logic is same in all the functions. Can we have a more generic function to avoid repeatation.